code/modules/mod/modules/modules_general.dm

```diff
@@ -128,40 +128,51 @@
 	overlay_state_inactive = "module_jetpack"
 	overlay_state_active = "module_jetpack_on"
 	/// Do we stop the wearer from gliding in space.
-	var/stabilizers = FALSE
+	var/stabilize = FALSE
+	var/thrust_callback
 
-/obj/item/mod/module/jetpack/proc/set_stabilizers(new_stabilizers)
-	if(stabilizers == new_stabilizers)
-		return
-	stabilizers = new_stabilizers
+/obj/item/mod/module/jetpack/Initialize(mapload)
+	. = ..()
+	thrust_callback = CALLBACK(src, PROC_REF(allow_thrust))
+	configure_jetpack(stabilize)
+
+/obj/item/mod/module/jetpack/Destroy()
+	thrust_callback = null
+	return ..()
+
+/**
+ * configures/re-configures the jetpack component
+ *
+ * Arguments
+ * stabilize - Should this jetpack be stabalized
+ */
+/obj/item/mod/module/jetpack/proc/configure_jetpack(stabilize)
+	src.stabilize = stabilize
+
+	AddComponent( \
+		/datum/component/jetpack, \
+		src.stabilize, \
+		COMSIG_MODULE_TRIGGERED, \
+		COMSIG_MODULE_DEACTIVATED, \
+		MOD_ABORT_USE, \
+		thrust_callback, \
+		/datum/effect_system/trail_follow/ion/grav_allowed \
+	)
 
 /obj/item/mod/module/jetpack/get_configuration()
 	. = ..()
-	.["stabilizers"] = add_ui_configuration("Stabilizers", "bool", stabilizers)
+	.["stabilizers"] = add_ui_configuration("Stabilizers", "bool", stabilize)
 
 /obj/item/mod/module/jetpack/configure_edit(key, value)
 	switch(key)
 		if("stabilizers")
-			set_stabilizers(value)
+			configure_jetpack(value)
 
 /obj/item/mod/module/jetpack/proc/allow_thrust()
-	if(!active)
```

Comment:
Stupid question, why not check for active here? Wouldn't the jetpack work without the modsuit being powered, or is there some other check?

Reply:
`allow_thrust` (i.e. the `thrust_callback` on the jetpack and which is passed as `check_on_move` to the component (yes it passes through 3 different but meaningful names unfortunately) is only invoked in `/datum/component/jetpack/proc/activate`, which is only called as a result of the `COMSIG_MODULE_TRIGGERED` signal firing in `/obj/item/mod/module/proc/on_activation`. And just before that, `active` is checked for us:

https://github.com/ParadiseSS13/Paradise/blob/5ca10363c1fedcf9a9c8326cc9cf612bd9f5eabc/code/modules/mod/modules/_modules.dm#L106-L110